api/src/main/java/io/druid/data/input/impl/DimensionSchema.java

```diff
@@ -26,10 +26,12 @@
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
 import com.fasterxml.jackson.annotation.JsonValue;
 import com.google.common.base.Preconditions;
+import io.druid.guice.annotations.PublicApi;
 import io.druid.java.util.common.StringUtils;
 
 /**
  */
+@PublicApi
```

Comment:
It's because ParseSpec is an extension point, and it returns DimensionsSpec, which is constructed using DimensionSchema. So implementors of custom ParseSpecs might need to use this class.